src/engine/TransitivePathImpl.h

```diff
@@ -319,24 +319,39 @@ class TransitivePathImpl : public TransitivePathBase {
    * @return cppcoro::generator<TableColumnWithVocab> An generator for
    * the transitive hull computation
    */
-  static cppcoro::generator<TableColumnWithVocab> setupNodes(
+  static ad_utility::InputRangeTypeErased<TableColumnWithVocab> setupNodes(
       const TransitivePathSide& startSide,
       std::shared_ptr<const Result> startSideResult) {
     if (startSideResult->isFullyMaterialized()) {
-      // Bound -> var|id
-      ql::span<const Id> startNodes = startSideResult->idTable().getColumn(
-          startSide.treeAndCol_.value().second);
-      co_yield TableColumnWithVocab{&startSideResult->idTable(), startNodes,
-                                    startSideResult->getCopyOfLocalVocab()};
-    } else {
-      for (auto& [idTable, localVocab] : startSideResult->idTables()) {
+      auto getter = [&startSide, startSideResult]() {
         // Bound -> var|id
-        ql::span<const Id> startNodes =
-            idTable.getColumn(startSide.treeAndCol_.value().second);
-        co_yield TableColumnWithVocab{&idTable, startNodes,
-                                      std::move(localVocab)};
-      }
+        ql::span<const Id> startNodes = startSideResult->idTable().getColumn(
+            startSide.treeAndCol_.value().second);
+        return ad_utility::LoopControl<TableColumnWithVocab>::breakWithValue(
+            TableColumnWithVocab{&startSideResult->idTable(), startNodes,
+                                 startSideResult->getCopyOfLocalVocab()});
+      };
+
+      return ad_utility::InputRangeTypeErased<TableColumnWithVocab>{
```

Comment:
at the beginning of the function, add a `using namespace ad_utility` to make these long names a little shorter.

Reply:
And do you need the explicit template argument  here (for the type-erased range), let me know if dropping it doesn't work, it should be possible.